flan/flan.c

```diff
@@ -0,0 +1,883 @@
+    }
+  }
+
+out:
+  return ret;
+
+}
+
+struct flan_oinfo* flan_get_oinfo(struct flan_handle *flanh, bool create)
+{
+  int ret = 0;
+  struct flan_oinfo *oinfo;
+
+  ret = flan_init_dirhandle(flanh);
+  if (ret)
+  {
+    printf("Error initializing dirhandle\n");
+    return NULL;
+  }
+
+  // We ran out of md objects
+  if(flan_dir.cur && !(flan_dir.cur % flan_oi_ob))
+    return NULL;
+
+  if (!((flan_dir.cur + 1) % flan_oi_ob))
+  {
+    oinfo = ((struct flan_oinfo *)flan_dir.buf + (flan_oi_ob - 1));
+    // Check if next entry is a pointer to another object
+    if (oinfo->size == UINT64_MAX)
+    {
+      flan_dir.fla_oh = oinfo->fla_oh;
+      fla_object_read(flanh->fs, flanh->ph, &oinfo->fla_oh, flan_dir.buf, 0, flan_obj_sz);
+      // Skip over the oinfo object which points to the next object used for MD
+      flan_dir.cur++;
+    }
+    else if(create)// Lets try to allocate a new object
+    {
+      struct fla_object new_md_obj;
+      // Grab a object from the pool, zero it out, and set it to root
+      ret = fla_object_create(flanh->fs, flanh->ph, &new_md_obj);
+      if (!ret)
+      {
+        oinfo->size = UINT64_MAX;
+        oinfo->fla_oh = new_md_obj;
+        fla_object_write(flanh->fs, flanh->ph, &flan_dir.fla_oh, flan_dir.buf, 0, flan_obj_sz);
+        memset((void *)flan_dir.buf, 0, flan_obj_sz);
+        flan_dir.fla_oh = new_md_obj;
+        // Skip over the oinfo object which points to the next MD object
+        flan_dir.cur++;
+      }
+      else
+      {
+        printf("Error allocating MD object bailing\n");
+        exit(0);
+      }
+    }
+  }
+
+  oinfo = ((struct flan_oinfo *)flan_dir.buf + (flan_dir.cur % flan_oi_ob));
+  flan_dir.cur++;
+  return oinfo;
+}
+
+int flan_object_create(const char *name, struct flan_handle *flanh, struct flan_oinfo *oinfo)
+{
+  int ret;
+  unsigned int namelen = strlen(name);
+  struct fla_object fla_obj;
+
+  if (namelen > FLAN_OBJ_NAME_LEN_MAX)
+  {
+    printf("flan obj create Name:%s longer than max len of:%d\n", name,  FLAN_OBJ_NAME_LEN_MAX);
+    return -EINVAL;
+  }
+
+  ret = fla_object_create(flanh->fs, flanh->ph, &fla_obj);
+  if (ret)
+  {
+    printf("Error allocating fla object\n");
+    return ret;
+  }
+
+  memcpy(oinfo->name, name, namelen + 1);
+  oinfo->size = 0;
+  oinfo->fla_oh = fla_obj;
+
+  if (!flanh->is_zns)
+  {
+    // TOOD make sure this is safe with object handle table now having private copies
+    ret = fla_object_write(flanh->fs, flanh->ph, &flan_dir.fla_oh, flan_dir.buf, 0, flan_obj_sz);
+    if (ret)
+    {
+      printf("Error writing md for object :%d\n", num_objects);
+      exit(0);
+    }
+  }
+  else
+    flanh->is_dirty = true;
+
+  num_objects++;
+
+  return ret;
+
+}
+
+struct flan_oinfo *flan_find_oinfo(struct flan_handle *flanh, const char *name)
+{
+  struct flan_oinfo *oinfo;
+
+  flan_reset_pool_dir();
+  while (((oinfo = flan_get_oinfo(flanh, false)) != NULL) &&
+         strncmp(oinfo->name, name, FLAN_OBJ_NAME_LEN_MAX));
```

Comment:
I believe that there is an error here.
You use basename for all the names but here you take the full name and compare it with the oinfo->name which has been previously passed through basename. 
1. A temporary fix is to compare the strings in reverse order and only considering the minimum length. This is the solution that I took and it seems to fix the issue of db_bench being able to restart
2. Another, more important questions is: why use basename?